add support for mongodb relations

ivangsa opened the pull request:
add support for mongodb relations

Fix jhipster/generator-jhipster#7944

Please make sure the below checklist is followed for Pull Requests.
  - [ ] [Travis tests](https://travis-ci.org/jhipster/jhipster-umk/pull_requests) are green
  - [x] Tests are added where necessary
  - [x] Documentation is added/updated where necessary
  - [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-core/blob/master/CONTRIBUTING.md) are followed

MathieuAA commented on lib/parser/entity_parser.js:
```diff
@@ -76,7 +76,7 @@ function checkNoSQLModeling(passedJdlObject, passedDatabaseType) {
     return;
   }
   if (passedJdlObject.getRelationshipQuantity() !== 0 && !DatabaseTypes.isSql(passedDatabaseType)) {
```
There's Cassandra too.

ivangsa replied:
Thanks, fixing it...

MathieuAA commented on test/spec/parser/entity_parser_test.js:
```diff
@@ -57,34 +57,34 @@ describe('EntityParser', () => {
       });
-      context('such as invalid databaseType', () => {
-        let jdlObject = null;
+      // context('such as invalid databaseType', () => {
```
Please remove the comments

ivangsa commented:
removed that test that expects an exception when parsing relations for mongodb
and updated the test that expects valid arguments for mongodb to include relationships as valid
